scripts/cleanup_datasets/pgcleanup.py

```diff
@@ -655,6 +662,121 @@ def zero_disk_usage(self):
         self.log.info('zero_disk_usage user_ids: %s', ' '.join(str(i) for i in user_ids))
 
 
+class PurgeDeletedUsersGDPR(PurgesHDAs, RemovesMetadataFiles, Action):
+    """
+    - Perform all steps in the PurgeDeletedUsers/purge_deleted_users action
+    - Obfuscate User.email and User.username for all users purged in this step.
+
+    NOTE: Your database must have the pgcrypto extension installed e.g. with:
+      CREATE EXTENSION IF NOT EXISTS pgcrypto;
+    """
+    _action_sql = """
+        WITH purged_user_ids
+          AS (     UPDATE galaxy_user
+                      SET purged = true,
+                          disk_usage = 0,
+                          email = encode(digest(email || '{epoch_time}', 'sha1'), 'hex'),
```

Comment:
we could just uuid it here, and then wouldn't need the pgcrypto extension. It doesn't need to be the same. (GDPR would probably encourage us that none of their data is in any way included.)

Reply:
@hexylena how about I do that in the dev branch followup? Only reason I did this was so that it works exactly the same as the purge function in Galaxy.

Reply:
sounds good. we can unify the implenentations then